src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -1958,7 +1980,7 @@ private String init(boolean initFull) {
                 return permissionsWrapper.notAuthorized();
             }
 
-            if (!retrieveDatasetVersionResponse.wasRequestedVersionRetrieved()) {
+            if (retrieveDatasetVersionResponse != null && !retrieveDatasetVersionResponse.wasRequestedVersionRetrieved()) {
```

Comment:
Is this null check necessary? I guess its being extra safe, but you do check for mull further up, no?

Reply:
This one IS actually necessary. Because in this PR I finally fixed the bug where the page was not working when called by `versionId=`. So yes, there can be a legit case where retrieveDatasetVersionResponse is going to be null here. 
(and I figured when the page gets called by the numeric id, if it doesn't exist, it should always be 404/not found, "no substitutes").